pipelines/Chat_gpt_de.yaml

```
@@ -0,0 +1,142 @@
+# This is a friendly editor that helps you create your pipelines with autosuggestions. To use them, press control + space on your keyboard.
+# Whenever you need to specify a model, this editor helps you out as well. Just type your Hugging Face organization and a forward slash (/) to see available models.
+
+# A RAG pipeline optimized for chat settings. It sends a summary of chat history, without including the documents every time.
+version: '1.23.0'
+
+# This section defines nodes that you want to use in your pipelines. Each node must have a name and a type. You can also set the node's parameters here.
+# The name is up to you, you can give your component a friendly name. You then use components' names when specifying their order in the pipeline.
+# Type is the class name of the component.
+components:
+  - name: DocumentStore
+    type: DeepsetCloudDocumentStore # The only supported document store in deepset Cloud
+    params:
+      embedding_dim: 768
+      similarity: cosine
+  - name: EmbeddingRetriever # Selects the most relevant documents from the document store
+    type: EmbeddingRetriever # Uses a Transformer model to encode the document and the query
+    params:
+      document_store: DocumentStore
+      embedding_model: PM-AI/bi-encoder_msmarco_bert-base_german # Model optimized for semantic search.
+      model_format: sentence_transformers
+      top_k: 20 # The number of results to return
+  - name: Reranker # Uses a cross-encoder model to rerank the documents returned by the two retrievers
+    type: CNSentenceTransformersRanker
+    params:
+      model_name_or_path: svalabs/cross-electra-ms-marco-german-uncased # Model optimized for reranking
+      top_k: 4 # The number of results to return
+      batch_size: 20  # Try to keep this number equal or larger to the sum of the top_k of the two retrievers so all docs are processed at once
+      model_kwargs:  # Additional keyword arguments for the model
+        torch_dtype: torch.float16
+  - name: qa_template
+    type: PromptTemplate
+    params:
+      output_parser:
+        type: AnswerParser
+      prompt: >
+        Du bist ein technischer Experte.
+        {new_line}Du beantwortest die Fragen wahrheitsgemäß auf Grundlage der vorgelegten Dokumente.
+        {new_line}Wenn die Antwort in mehreren Dokumenten enthalten ist, fasse diese zusammen.
+        {new_line}Ignorierst du Dokumente, die keine Antwort auf die Frage enthalten.
+        {new_line}Antworte nur auf der Grundlage der vorgelegten Dokumente. Erfinde keine Fakten.
+        {new_line}Verwende immer Verweise in der Form [NUMMER DES DOKUMENTS], wenn du Informationen aus einem Dokument verwendest, z. B. [3] für Dokument[3].
+        {new_line}Der Verweis darf sich nur auf die Nummer beziehen, die in eckigen Klammern hinter der Passage steht.
+        {new_line}Andernfalls verwende in deiner Antwort keine Klammern und gib NUR die Nummer der Textstelle an, ohne das Wort Textstelle zu erwähnen.
```

```suggestion
        {new_line}Andernfalls verwende in deiner Antwort keine Klammern und gib NUR die Nummer des Dokuments an, ohne das Wort Dokument zu erwähnen.
```

I'd try to avoid referencing documents with different terms